File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02917/p02917_num4_parsed.py
def main():
    # Read the input values
    N = int(input())
    B = list(map(int, input().split()))

    # Calculate the maximum possible sum of the elements of A
    A = [0] * (N + 1)
    A[0] = B[0]
    for i in range(1, N):
        A[i] = min(B[i-1], B[i])
    A[N] = B[N-1]

    max_sum = sum(A)

    # Print the maximum possible sum of the elements of A
    print(max_sum)

if __name__ == '__main__':
    main()

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02917/p02917_num4_parsed.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int main() {
    int N;
    std::cin >> N;
    
    std::vector<int> B(N-1);
    for (int i = 0; i < N-1; i++) {
        std::cin >> B[i];
    }
    
    std::vector<int> A(N);
    A[0] = B[0];
    for (int i = 1; i < N-1; i++) {
        A[i] = std::min(B[i-1], B[i]);
    }
    A[N-1] = B[N-2];
    
    int sum = 0;
    for (int i = 0; i < N; i++) {
        sum += A[i];
    }
    
    std::cout << sum << std::endl;
    
    return 0;
}